File: packages/js-dpp/lib/stateTransition/errors/StateTransitionIsNotSignedError.js
const DPPError = require('../../errors/DPPError');

class StateTransitionIsNotSignedError extends DPPError {
  /**
   *
   * @param {AbstractStateTransition} stateTransition
   */
  constructor(stateTransition) {
    super('State Transition is not signed');

    this.stateTransition = stateTransition;
  }

  /**
   * Get unsigned state transition
   *
   * @return {AbstractStateTransition}
   */
  getStateTransition() {
    return this.stateTransition;
  }
}

module.exports = StateTransitionIsNotSignedError;

File: packages/rs-dpp/src/state_transition/errors/state_transition_is_not_signed_error.rs
use thiserror::Error;

use crate::state_transition::StateTransition;
use crate::errors::ProtocolError;

#[derive(Error, Debug, Clone)]
#[error("State Transition is not signed")]
#[ferment_macro::export]
pub struct StateTransitionIsNotSignedError {
    pub state_transition: StateTransition,
}

impl StateTransitionIsNotSignedError {
    pub fn new(state_transition: StateTransition) -> Self {
        Self { state_transition }
    }

    pub fn state_transition(&self) -> StateTransition {
        self.state_transition.clone()
    }
}

impl From<StateTransitionIsNotSignedError> for ProtocolError {
    fn from(err: StateTransitionIsNotSignedError) -> Self {
        Self::StateTransitionIsNotSignedError(err)
    }
}
